wordpress-mobile/WordPress-iOS
# [Site Intent] Clear custom verticals between VC loads

Fixes an issue where custom verticals on the Site Intent screen could be retained between site creation flows.

| Before | After |
| -------| ----- |
| <video src="https://user-images.githubusercontent.com/2092798/162228674-ebccbfd9-767f-4977-8b44-462b16cbff36.mp4" /> | <video src="https://user-images.githubusercontent.com/2092798/162229242-22d5c0f8-f056-4aac-8803-d0f2470457c6.mp4" /> |

## To test:

**Prerequisites:** The user must be in the `treatment` group for Site Intent and the feature flag must be enabled. [More information here](https://github.com/wordpress-mobile/WordPress-iOS/pull/18083). To programmatically control the group, return `.treatment` [here](https://github.com/wordpress-mobile/WordPress-iOS/blob/bug/clear-custom-vertical/WordPress/Classes/ViewRelated/Site%20Creation/Wizard/SiteIntentAB.swift#L26).

1. Start the site creation flow by creating a new site
2. On the Site Intent view, type into the input to create a custom vertical name
3. Cancel, Skip, or select the custom vertical to finish the site creation flow
4. Start the site creation flow again
5. Tap on the input field
6. Observe that no custom vertical from the last flow was retained

## Regression Notes
1. Potential unintended areas of impact
- The Site Creation flow
- Selecting verticals
- Sending events related to Site Intent

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Choosing an existing vertical
- Choosing a custom vertical
- Completing the site creation flow

3. What automated tests I added (or what prevented me from doing so)
`SiteIntentDataTests` test outputs for given search inputs.

PR submission checklist:
